File: SceytChatUiKit/src/main/java/com/sceyt/sceytchatuikit/media/audio/AudioRecorder.java
package com.sceyt.sceytchatuikit.media.audio;


import com.sceyt.sceytchatuikit.media.DurationCallback;

public interface AudioRecorder {
    // Constructor accepts created File or file path

    boolean startRecording(Integer bitrate, DurationCallback durationCallback);

    void stopRecording();

    Integer getRecordingDuration();

    Integer[] getRecordingAmplitudes();
}

File: SceytChatUiKit/src/main/java/com/sceyt/sceytchatuikit/media/audio/AudioRecorder.kt
package com.sceyt.sceytchatuikit.media.audio

import com.sceyt.sceytchatuikit.media.DurationCallback

interface AudioRecorder {
    // Constructor accepts created File or file path
    fun startRecording(bitrate: Int, durationCallback: DurationCallback?): Boolean
    fun stopRecording()
    fun getRecordingDuration(): Int
    fun getRecordingAmplitudes(): Array<Int>
}